src/content/components/IdleSearchField.js

```diff
@@ -51,23 +44,45 @@ class IdleSearchField extends Component {
     }
   }
 
+  _onClearButtonClick() {
+    this.setState({ value: '' });
+    this._notifyIfChanged('');
+  }
+
+  _onClearButtonFocus(e) {
+    // prevent the focus on the clear button
+    if (e.relatedTarget) {
+      e.relatedTarget.focus();
+    } else {
+      e.target.blur();
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.defaultValue !== this.props.defaultValue) {
+      this._notifyIfChanged(nextProps.defaultValue || '');
+      this.setState({
+        value: nextProps.defaultValue || '',
+      });
+    }
+  }
+
   render() {
-    const { className, title, defaultValue } = this.props;
+    const { className, title } = this.props;
     return (
-      <span className={classNames('idleSearchField', className)}>
-        <input type='search'
+      <form className={classNames('idleSearchField', className)} onSubmit={e => e.preventDefault()}>
```

Comment:
TBH this is what stays from previous iterations of this patch:
1. trying to make the `type="reset"` below to automatically reset the input -- but this doesn't work because of how React sets the attribute `value`
2. then I used it to react to the reset click event, using the reset input's `form` property to jump back to the input. (something like `e.currentTarget.form.search`) but then this approach got reset (pun intended) by solely controlling the input through the state. You can also see the remain with the `name="search"` below.

Now I think this brings something for accessibility and to structure the page, instead of div/span soup :) as a form contains more semantic value than a span. The downside is that we need to cancel the submission.